pkg/ccl/streamingccl/replicationtestutils/testutils.go

```diff
@@ -154,25 +160,40 @@ func (c *TenantStreamingClusters) init() {
 	c.DestSysSQL.Exec(c.T, `SET CLUSTER SETTING cross_cluster_replication.enabled = true;`)
 }
 
-// StartDestTenant starts the destination tenant and returns a cleanup
-// function that shuts tenant SQL instance and closes all sessions.
-// This function will fail the test if ran prior to the Replication stream
-// closing as the tenant will not yet be active
-func (c *TenantStreamingClusters) StartDestTenant(ctx context.Context) func() error {
-	c.DestSysSQL.Exec(c.T, `ALTER TENANT $1 START SERVICE SHARED`, c.Args.DestTenantName)
-
-	destTenantConn := c.DestCluster.Server(0).SystemLayer().SQLConn(c.T, "cluster:"+string(c.Args.DestTenantName)+"/defaultdb")
+// StartDestTenant starts the destination tenant and returns a cleanup function
+// that shuts tenant SQL instance and closes all sessions. This function will
+// fail the test if ran prior to the Replication stream closing as the tenant
+// will not yet be active. If the caller passes withTestingKnobs, the
+// destination tenant starts up via a testServer.StartSharedProcessTenant().
+func (c *TenantStreamingClusters) StartDestTenant(
+	ctx context.Context, withTestingKnobs *base.TestingKnobs,
+) func() error {
+	if withTestingKnobs != nil {
+		var err error
+		_, c.DestTenantConn, err = c.DestCluster.Server(0).StartSharedProcessTenant(ctx, base.TestSharedProcessTenantArgs{
+			TenantID:    c.Args.DestTenantID,
+			TenantName:  c.Args.DestTenantName,
+			Knobs:       *withTestingKnobs,
+			UseDatabase: "defaultdb",
+		})
+		require.NoError(c.T, err)
+	} else {
+		c.DestSysSQL.Exec(c.T, `ALTER TENANT $1 START SERVICE SHARED`, c.Args.DestTenantName)
+		c.DestTenantConn = c.DestCluster.Server(0).SystemLayer().SQLConn(c.T, "cluster:"+string(c.Args.DestTenantName)+"/defaultdb")
+	}
```

Comment:
alternatively we could always call StartSharedProcessTenant

Reply:
I'd like to exercise that by default only executing `ALTER TENANT $1 START SERVICE SHARED` is sufficient for startup, rather than whatever occurs in StartSharedProcessTenant.